metpy/calc/indices.py

```diff
@@ -51,3 +52,64 @@ def precipitable_water(dewpt, p, top=400 * units('hPa')):
     pw = -1. * (np.trapz(w.magnitude, pres_layer.magnitude) * (w.units * pres_layer.units) /
                 (g * rho_l))
     return pw.to('millimeters')
+
+
+@exporter.export
+@check_units('[speed]', '[speed]', '[pressure]', '[length]', '[length]', '[length]')
+def mean_wind_pressure_weighted(u, v, p, hgt, top, bottom=None, interp=False):
```

Comment:
I think I'd go with bottom, depth as arguments so be consistent with `get_layer` and other layer based functions.